### DISCUSSION TITLE
Can't get Prettier to work properly with Angular templates

### DISCUSSION THREAD

**Discussion by asmolenic** [LazyVim/LazyVim#4676] [Q&A] [answered]
Hi, I'm attempting a switch from VS Code to Neovim/LazyVim and so far its not been very hard to configure. I work with Angular so I need to tweak LazyVim for that.

My steps so far:
1. Install Neovim 0.11 (nightly).
2. Install LazyVim (clone the repo in the right location on my drive)
3. Installed **ripgrep** and a **C** compiler
4. Startup Neovim in my project's root directory and wait for stuff to be installed.
5. I've repeated the previous step 2 more times for good measure (to ensure it finishes installing everything)
6. I've used ":TSInstall angular" to install angular language support for treesitter. Confirmed installation with ":TSInstallInfo"
7. I've use the :LazyExtras comand to install the required extras and restarted Neovim. After this step, Git showed me that the `lazyvim.json` file has been changed to:
``` json
{
  "extras": [
    "lazyvim.plugins.extras.lang.angular",
    "lazyvim.plugins.extras.lang.json",
    "lazyvim.plugins.extras.lang.markdown",
    "lazyvim.plugins.extras.lang.typescript",
    "lazyvim.plugins.extras.formatting.prettier"
  ],
  "news": {
    "NEWS.md": "6520"
  },
  "version": 6
}
```
I have confirmed that the Typescript and Angular LSPs are working properly. Automatic formatting (`<leader>cf`) seems to be working well on *.ts and *.scss files but it doesn't work on *.html files (I get the "No formatter available" error).

Running the :ConformInfo command while on a *.html buffer yields the following output:
```
Log file: C:\Users\asmol\AppData\Local\nvim-data\conform.log

Formatters for this buffer:
<none>

Other formatters:
fish_indent unavailable: Command not found
markdown-toc unavailable: Condition failed
markdownlint-cli2 unavailable: Condition failed
prettier ready (markdown, typescriptreact, css, graphql, handlebars, less, markdown.mdx, scss, json, vue, typescript, html, javascript, yaml, jsonc, javascriptreact) D:\MyProject\node_modules\.bin\prettier.cmd
shfmt ready (sh) C:\Users\asmol\AppData\Local\nvim-data\mason\bin\shfmt.CMD
stylua ready (lua) C:\Users\asmol\AppData\Local\nvim-data\mason\bin\stylua.CMD
```

Ran :checkhealth and got the following conform.nvim report:
```
conform.nvim report ~
- Log file: C:\Users\asmol\AppData\Local\nvim-data\conform.log
- WARNING fish_indent unavailable: Command not found
- WARNING markdown-toc unavailable: Condition failed
- WARNING markdownlint-cli2 unavailable: Condition failed
- WARNING prettier unavailable: Condition failed
- OK shfmt ready (sh)
- OK stylua ready (lua)
```

So, to my limited understanding of how Neovim works, it seems that Prettier is not attaching to *.html buffers. What am I missing?

Thanks a bunch,
Andrei

**Comment by dpetka2001** [comment]
What does `:set ft?` show in the html file that you have problems? My guess is Neovim recognizes the filetype as not `html`.

**Reply by asmolenic** [reply]
You are correct. It recognizes it as htmlangular, most likely because of the lazyvim.plugins.extras.lang.angular "extra".

I guess I should somehow tell Prettier to keep an eye on files of type htmlangular as well. Am I on the right track?

Thanks!

**Reply by dpetka2001** [reply, marked as answer]
You could try the following in `~/.config/nvim/lua/plugins/conform.lua`
```lua
return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        htmlangular = { "prettier" },
      },
    },
  },
}
```
Take a look at the docs both on LazyVim website and `conform.nvim` github repo for more info on how to configure it if you want anything more than that.

**Reply by asmolenic** [reply]
It worked as a charm, thanks!
